rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,538 @@
+
+### Trip arrived
+Inspectors are required to note when trains arrive at their location. This event is generated by that confirmation, and includes the relevant information about who performed the trip. 
+
+Event type: `com.mbta.ctd.glides.trip-arrived`
+Fields in the event:
+- `author` (Author): the inspector adding the trip.
+- `tripKey` (Trip Key): which trip is being confirmed.
+- `comment` (string, optional): free text information about the trip.
+- `location` (Location): the location at which the train has arrived.
+- `arrivalTime` (Time): if present, the time that the train was recorded arriving at the specified location.
+- `consist` (array of Car): the cars which performed this trip.
+- `operators` (array of Operator): the list of operators who to performed this trip.Even if the scheduled operators performed the trip, those values MUST be included here.
+
+*Notes*
+- `consist` and `operators` MUST be the same length; if they are not, the behavior of consumers is undefined.
+
+### Trip departed
+Inspectors are required to note when trains arrive at their location. This event is generated by that confirmation, and includes the relevant information about who performed the trip. 
+
+Event type: `com.mbta.ctd.glides.trip-departed`
+Fields in the event:
+- `author` (Author): the inspector adding the trip
+- `tripKey` (Trip Key): which trip is being confirmed
+- `comment` (string, optional): free text information about the trip
+- `location` (Location): the location from which the train has departed.
+- `departureTime` (Time): the time that the train was recorded departing from the specified location.
+- `consist` (array of Car): the cars which performed this trip.
+- `operators` (array of Operator): the list of operators who to performed this trip. Even if the scheduled operators performed the trip, those values MUST be included here.
+
+*Notes*
+- `consist` and `operators` MUST be the same length; if they are not, the behavior of consumers is undefined.
+
+### Trip dropped
+This removes a trip (either added or scheduled), indicating that it will not be running.
+
+Event type: `com.mbta.ctd.glides.trip-dropped`
+Fields in the event:
+- `author` (Author): the inspector dropping the trip.
+- `tripKey` (Trip Key): which trip to drop. can be either an added or scheduled trip.
+- `reason` (string): a text description of why the trip is being dropped.
+
+### Trip restored
+This restores a trip (either added or scheduled) that was previously dropped.
+
+Event type: `com.mbta.ctd.glides.trip-restored`
+Fields in the event:
+- `author` (Author): the inspector restoring the trip.
+- `tripKey` (Trip Key): which trip to restore. can be either an added or scheduled trip.
+
+### Trip updated
+This indicates that a trip was updated in some fashion before the train has departed or arrived at the location:  consist, operators, departure time. Fields which are not present are not considered to be updated.
```

Comment:
Updates can be done after the fact, too. If the inspector is busy (for example during a shift change) they might record information 30-60 minutes after the trip departs/arrives.